src/quemb/shared/external/optqn.py

```diff
@@ -388,7 +388,6 @@ def get_atbe_Jblock_frag(fobj, res_func):
 
 def get_be_error_jacobian_selffrag(self, jac_solver="HF"):
     Jes = [None] * self.Nfrag
-    Jcs = [None] * self.Nfrag
     xes = [None] * self.Nfrag
     xcs = [None] * self.Nfrag
     ys = [None] * self.Nfrag
```

Comment:
if you know the correct datatype (i.e. float, int, or complex) then it would be great to also replace the `[None] * self.Nfrag` with `np.empty` calls.

If you are not sure about it, then we just leave it as is.

Reply:
Hmm I do know the datatypes, but do not know what their sizes are off the top of my head :(

Reply:
Shouldn't the size just be `self.Nfrag`?
i.e. 
```python3
np.empty(self.Nfrag, dtype=??)
```

Reply:
The elements of the `Jes`, for example, would be numpy arrays, which I do not know the size of. Do you mean I should do `np.empty(self.Nfrag, dtype=np.ndarray`)?

Reply:
Summarizing the discussion with @mscho527 : 

with ragged arrays it becomes too complicated now. let's postpone it for later